### PR TITLE
test: Testcases for Submit command value when clicked, if available

### PR DESCRIPTION
**Pull request by VijithaEkanayake**
testcases for https://github.com/h2oai/wave/issues/661

### CODE REVIEW

File: ui/src/toolbar.test.tsx [hunk 1]
```diff
@@ -13,22 +13,60 @@
...
-import { render } from '@testing-library/react'
+import { fireEvent, render } from '@testing-library/react'
 import { View } from './toolbar'
 import * as T from './qd'
 
 const
   name = 'toolbar',
+  commandName = 'toolbar_command',
+  commandValue = 'toolbar_command_value',
+  commandLabel = commandName,
```
Comment (mturoci) [inline ui/src/toolbar.test.tsx:24]: Not needed as it is the name.

File: ui/src/toolbar.test.tsx [hunk 2]
```diff
@@ -13,22 +13,60 @@
...
+  toolbarPropsWithValueAttr: T.Card<any> = {
+    name,
+    state: { items: [{ name: commandName, value: commandValue, label: commandLabel }] },
+    changed: T.box(false)
+  },
+  toolbarPropsWithoutValueAttr: T.Card<any> = {
+    name,
+    state: { items: [{ name: commandName, label: commandLabel }] },
+    changed: T.box(false)
   }
```
Comment (mturoci) [inline ui/src/toolbar.test.tsx:39]: No need for these, use existing `toolbarProps` with base state: ` state: { items: [{ name: commandName, label: commandLabel }] },` and add value within specific test case as needed. Be careful to not change the original base props instance for further tests.

File: ui/src/toolbar.test.tsx [hunk 3]
```diff
@@ -13,22 +13,60 @@
...
+  })
 
   it('Renders data-test attr', () => {
     const { queryByTestId } = render(<View {...toolbarProps} />)
     expect(queryByTestId(name)).toBeInTheDocument()
   })
-})
\ No newline at end of file
+
+  it('Sets args and calls sync on click - with value attr', () => {
```
Comment (mturoci) [inline ui/src/toolbar.test.tsx:52]: missing check if sync was called

File: ui/src/toolbar.test.tsx [hunk 4]
```diff
@@ -13,22 +13,60 @@
...
+    const syncMock = jest.fn()
+    T.qd.sync = syncMock
+
+    const {getByText} = render(<View {...toolbarPropsWithValueAttr} />)
+    fireEvent.click(getByText(commandLabel))
+
+    expect(T.qd.args[commandName]).toBe(commandValue)
+  })
+
+  it('Sets args and calls sync on click - without value attr', () => {
```
Comment (mturoci) [inline ui/src/toolbar.test.tsx:62]: missing check if sync was called

**Review by mturoci — CHANGES_REQUESTED** [review]
Thanks @VijithaEkanayake. A few comments. Please also include tests for case when value starts with a hash.

**Comment by VijithaEkanayake** [comment]
> Thanks @VijithaEkanayake. A few comments. Please also include tests for cases when the value starts with a hash.

@mturoci All the suggestions are introduced. Kindly have a look.

**Comment by VijithaEkanayake** [comment]
@mturoci  Please refer to PR: https://github.com/h2oai/wave/pull/717 which introduces the requested changes. I don't seem to have the permissions to request a review from anyone it just default reviewer.

**Pull request closed without merging**